### PR TITLE
Add types for node-hrx

### PR DESCRIPTION
**Pull request by stof**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

If adding a new definition:
- [x] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [x] If this is for an npm package, match the name. If not, do not conflict with the name of an npm package.
- [x] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [x] Represents shape of module/library [correctly](https://www.typescriptlang.org/docs/handbook/declaration-files/library-structures.html)
- [x] `tslint.json` [should contain](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#linter-tslintjson) `{ "extends": "dtslint/dt.json" }`, and no additional rules.
- [x] `tsconfig.json` [should have](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#tsconfigjson) `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

### CODE REVIEW

File: types/node-hrx/index.d.ts [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+    readonly body: string;
+    readonly comment?: string;
+    isDirectory(): this is Directory;
+    isFile(): this is File;
+}
+
+export type HrxItem = File | Directory;
+
+export class ParseProblem extends Error {
+    constructor(message: string, position: { line: number, col: number });
```
Comment (peterblazejewicz) [inline types/node-hrx/index.d.ts:40]: this requires `details` props, doesn't?
https://github.com/issa-tseng/node-hrx/blob/c98c758139171e9bc810dc61f468b66463a21477/lib/util/problem.js#L5

File: types/node-hrx/index.d.ts [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+    readonly line: number;
+    readonly col: number;
+}
+
+export class LexicalProblem extends Error {
+    constructor(message: string, line: number);
```
Comment (peterblazejewicz) [inline types/node-hrx/index.d.ts:46]: same here, optonial `details`:
https://github.com/issa-tseng/node-hrx/blob/c98c758139171e9bc810dc61f468b66463a21477/lib/util/problem.js#L14

File: types/node-hrx/index.d.ts (hunk 1 above)
Comment (stof) [reply types/node-hrx/index.d.ts:40]: hmm, indeed. I missed it (not that the library seems to never populate them for ParseProblem, so I don't really know how to describe it

Comment (peterblazejewicz) [reply types/node-hrx/index.d.ts]: no worries, we, or someone using this, can refine this later

File: types/node-hrx/index.d.ts [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+    readonly body: string;
+    readonly comment?: string;
+    isDirectory(): this is Directory;
+    isFile(): this is File;
+}
+
+export type HrxItem = File | Directory;
+
+export class ParseProblem extends Error {
+    constructor(message: string, position: { line: number, col: number }, details?: {});
```
Comment (peterblazejewicz) [inline types/node-hrx/index.d.ts:40]:
```suggestion
    constructor(message: string, position: { line: number, col: number }, details?: Record<string, unknown>);
```

File: types/node-hrx/index.d.ts [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+    readonly line: number;
+    readonly col: number;
+    readonly details: {};
+}
+
+export class LexicalProblem extends Error {
+    constructor(message: string, line: number, details?: { path?: string });
```
Comment (peterblazejewicz) [inline types/node-hrx/index.d.ts:47]:
```suggestion
    constructor(message: string, line: number, details?: { path?: string, [key: string]: unknown });
```

File: types/node-hrx/index.d.ts (hunk 1 above)
Comment (peterblazejewicz) [reply types/node-hrx/index.d.ts]: I've added type declared as opaque dictionary, with one know `path` property, should fit any scenario for now. Let me know if that is OK for you

**Review by peterblazejewicz — APPROVED** [review]
LGTM!
@stof thx!

**Comment by stof** [comment]
Ready to merge

**Pull request merged**